go/parquet/pqarrow/file_writer.go

```diff
@@ -134,6 +134,13 @@ func (fw *FileWriter) RowGroupTotalBytesWritten() int64 {
 	return 0
 }
 
+// WriteBuffered allows to write records and decide where to break your row group
+// based on the TotalBytesWritten rather than on the max row group len.
+// If using Records, this should be paired with NewBufferedRowGroup,
+// while Write will always write a new record as a row group in and of itself.
+//
+// Performance-wise WriteBuffered might be more favorable than Write
+// especially if dealing with lots of records that have only a small amount of rows.
```

Comment:
Mention that the tradeoff is that more memory will be utilized to keep the whole row group buffered in memory before it starts writing (since Parquet files must write an entire column before writing the next column).

Reply:
Done in 9c4ca94137a143986e69feb07a0ab94baee7e334